### ISSUE TITLE
Create integration test to check salable qty for product

### ISSUE THREAD

**Issue by smoskaluk** [magento/inventory#2064]
### Preconditions

1. Create  simple product with source item qty: 0,  source status: IN_STOCK, backorder - enabled, use decimal - On, out of stock treshold = -3.

### Steps to reproduce

1.Check programmtically \Magento\InventorySalesApi\Api\GetProductSalableQtyInterface

### Expected result

1.  Salable qty = 3

### Actual result

1. . Salable qty = 0

**Comment by maghamed** [comment]
in the scope of this ticket, we need to provide an Integration test for 
```

namespace Magento\InventorySalesApi\Api;

/**
 * Service which returns Quantity of products available to be sold by Product SKU and Stock Id.
 * This service calculates the salable qty taking into account existing reservations for
 * given sku and stock id and subtracting min qty (a.k.a. "Out-of-Stock Threshold")
 *
 * @api
 */
interface GetProductSalableQtyInterface
{
    /**
     * Get Product Quantity for given SKU and Stock
     *
     * @param string $sku
     * @param int $stockId
     * @return float
     * @throws \Magento\Framework\Exception\InputException
     * @throws \Magento\Framework\Exception\LocalizedException
     */
    public function execute(string $sku, int $stockId): float;
}
```

and cover the case when Backorder and Minimal Threshold is set

**Comment by josh-carter** [comment]
@smoskaluk @maghamed I've created a PR in scope of this issue. However, there doesn't seem to be a problem - https://github.com/magento-engcom/msi/pull/2074

I've added a new Product/StockItem to the existing integration test for GetProductSalableQtyInterface, the product has the following stock data:

```
'qty' => 0,
'is_in_stock' => true,
'manage_stock' => true, 
'is_qty_decimal' => true,
'min_qty' => -3,
'backorders' => true
```

Expected result is 3, which is matched by the outcome of the test.

@smoskaluk perhaps theres another aspect to this missed off the issue? Let me know and i'll proceed accordingly.
